beacon-chain/slasher/detect_attestations.go

```diff
@@ -27,51 +27,55 @@ func (s *Service) checkSlashableAttestations(
 	if err != nil {
 		return nil, errors.Wrap(err, "could not check slashable double votes")
 	}
+
 	log.WithField("elapsed", time.Since(start)).Debug("Done checking double votes")
+
 	slashings = append(slashings, doubleVoteSlashings...)
 
 	// Surrounding / surrounded votes
-	groupedAtts := s.groupByValidatorChunkIndex(atts)
-	log.WithField("numBatches", len(groupedAtts)).Debug("Batching attestations by validator chunk index")
-	start = time.Now()
-	batchTimes := make([]time.Duration, 0, len(groupedAtts))
+	groupedByValidatorChunkIndexAtts := s.groupByValidatorChunkIndex(atts)
+	log.WithField("numBatches", len(groupedByValidatorChunkIndexAtts)).Debug("Batching attestations by validator chunk index")
+	groupsCount := len(groupedByValidatorChunkIndexAtts)
+	batchDurations := make([]time.Duration, 0, len(groupedByValidatorChunkIndexAtts))
```

Comment:
batchDurations is only used to calculate the totalBatchDuration. Why not just accumulate / update totalBatchDuration and skip this slice allocation?

Reply:
Fixed in this commit: https://github.com/prysmaticlabs/prysm/pull/13589/commits/7acbc4d8a2465ee369a442f532e591f584dca297